gmso/formats/lammpsdata.py

```diff
@@ -183,48 +183,159 @@ def write_lammpsdata(topology, filename, atom_style="full"):
+                    / simplify(param.expression)
+                ).is_real:
                     data.write(
                         "{}\t{:.5f}\t{:.5f}\n".format(
                             idx + 1,
-                            angle_type.parameters["k"]
-                            .in_units(u.Unit("kcal/mol/radian**2"))
+                            param.parameters["epsilon"]
+                            .in_units(u.Unit("kcal/mol"))
                             .value
-                            / 2,
-                            angle_type.parameters["theta_eq"]
-                            .in_units(u.Unit("degree"))
+                            / scaling_factor,
+                            param.parameters["sigma"]
+                            .in_units(u.angstrom)
                             .value,
                         )
                     )
+                else:
+                    raise ValueError(
+                        'Pair Style "{}" is invalid or is not currently supported'.format(
+                            param.expression
+                        )
+                    )
+            if topology.bonds:
+                data.write("\nBond Coeffs\n\n")
+                for idx, bond_type in enumerate(topology.bond_types):
 
+                    # expected harmonic potential expression for lammps
+                    bond_expression = "k * (r-r_eq)**2"
+
+                    if sympify(bond_type.expression).equals(
+                        sympify(bond_expression)
+                    ):
+                        data.write(
+                            "{}\t{:.5f}\t{:.5f}\n".format(
+                                idx + 1,
+                                bond_type.parameters["k"]
+                                .in_units(u.Unit("kcal/mol/angstrom**2"))
+                                .value,
+                                bond_type.parameters["r_eq"]
+                                .in_units(u.Unit("angstrom"))
+                                .value,
+                            )
+                        )
+                    elif sympify(bond_type.expression).equals(
+                        sympify("0.5 * k * (r-r_eq)**2")
+                    ):
+                        data.write(
+                            "{}\t{:.5f}\t{:.5f}\n".format(
+                                idx + 1,
+                                bond_type.parameters["k"]
+                                .in_units(u.Unit("kcal/mol/angstrom**2"))
+                                .value
+                                / 2.0,
+                                bond_type.parameters["r_eq"]
+                                .in_units(u.Unit("angstrom"))
+                                .value,
+                            )
+                        )
+                    elif (
+                        scaling_factor := simplify(bond_expression)
+                        / simplify(bond_type.expression)
+                    ).is_real:
```

Comment:
isn't the middle `elif` already covered by the next `elif` (where the scaling factor is calculated)

Reply:
Yes, it is covered.  The scaling_factor calculation would work for all cases (I did the explicit way initially, then did the more general form in case someone defined these parameters in a non-standard, but otherwise equivalent form).  We could just switch completely to the scaling_factor.